### DISCUSSION TITLE
Nx monorepo implementation

### DISCUSSION THREAD

**Discussion by holisticxyz** [wxt-dev/wxt#572] [Q&A]
I'm wondering if anyone has some examples on how to implement `wxt` inside an Nx monorepo?

**Comment by aklinker1** [comment]
There's nothing special when setting it up. Are you using a package-based monorepos or integrated?

The main thing that will probably confuse you is setting up typescript. TS config files can extend multiple other files, so make sure you keep extending `.wxt/tsconfig.json`.

If you need to setup path aliases, like `@shared` pointing to `packages/shared`, but them in your `wxt.config.ts` file, don't set them manually in the tsconfig file.

You may also need to disable auto-imports.

Hopefully that will get you started. LMK if you have other problems.

Here's a PNPM monorepo, a package-based one, as an example:

https://github.com/anime-skip/player

**Comment by spookyuser** [comment]
Will just add that if you're following some turborepo examples with nextjs and such, they will often put all the typescript config in it's own little package like `/packages/typescript-config`- if you're following that pattern extend from the monorepo tsconfig like they do in their examples and  and then `include:[` the .wxt folder in the tsconfig for the app so like

include: [ "./.wxt/**/*"] (don't forget the two dots ././)

This will give you the generated types but nothing else from the wxt tsconfig so you need to make sure that your tsconfig will be able to do what the generated one is doing

**Reply by aklinker1** [reply]
My recommendation here would be to extend both your shared config and the generated one inside `.wxt`.

```json
{
  "extends": ["tsconfig/base.json", "./.wxt/tsconfig.json"],
}
```

**Reply by spookyuser** [reply]
Thanks! I tried this I'm not sure why but it just didn't seem to work, will give it another shot though!
